k9mail/src/test/java/com/fsck/k9/activity/MessageReferenceTest.java

```diff
@@ -80,25 +82,30 @@ public void checkMessageReferenceWithChangedFlag() {
     }
 
     @Test
-    public void parseIdentityStringContainingBadVersionNumber() throws MessagingException {
-        MessageReference messageReference = new MessageReference("@:byBoYWkh:Zm9sZGVy:MTAxMDEwMTA=:ANSWERED");
+    public void parseIdentityStringContainingBadVersionNumber() {
+        MessageReference messageReference = MessageReference.parse("@:byBoYWkh:Zm9sZGVy:MTAxMDEwMTA=:ANSWERED");
 
-        assertNull(messageReference.getAccountUuid());
+        assertNull(messageReference);
     }
 
-    @Test(expected = MessagingException.class)
-    public void parseNullIdentityString() throws MessagingException {
-        new MessageReference(null);
+    @Test
+    public void parseNullIdentityString() {
+        MessageReference messageReference = MessageReference.parse(null);
+
+        assertNull(messageReference);
     }
 
-    @Test(expected = MessagingException.class)
-    public void parseIdentityStringWithCorruptFlag() throws MessagingException {
-        new MessageReference("!:%^&%^*$&$by&(BYWkh:Zm9%^@sZGVy:MT-35#$AxMDEwMTA=:ANSWE!RED");
+    @Test
+    public void parseIdentityStringWithCorruptFlag() {
+        MessageReference messageReference =
+                MessageReference.parse("!:%^&%^*$&$by&(BYWkh:Zm9%^@sZGVy:MT-35#$AxMDEwMTA=:ANSWE!RED");
+
+        assertNull(messageReference);
     }
 
     @Test
     public void equalsWithAnObjectShouldReturnFalse() {
-        MessageReference messageReference = new MessageReference(null, null, null, null);
+        MessageReference messageReference = new MessageReference("a", "b", "c", null);
```

Comment:
You're passing in values instead of null because you added pre-conditions - is it worth testing the existence of the pre-conditions? I couldn't see any.